src/main/java/com/javachallengers/simpson/controller/RestControllerExceptionHandler.java

```diff
@@ -0,0 +1,35 @@
+package com.javachallengers.simpson.controller;
+
+import java.util.Date;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.server.ResponseStatusException;
+
+import com.javachallengers.simpson.model.dto.ApiErrorDTO;
+
+@ControllerAdvice
+public class RestControllerExceptionHandler {
+	private static final Logger logger = LoggerFactory.getLogger(RestControllerExceptionHandler.class);
+	
+	@ExceptionHandler(value = { ResponseStatusException.class })
+	public ResponseEntity<ApiErrorDTO> handleResponseStatusException(ResponseStatusException ex) {
+		logger.error(ex.getMessage(), ex);
+		
+		String error = ex.getStatus().getReasonPhrase();
```

Comment:
Great, I like it! Is this code going to handle every Exception and send a clear message to the user?

Reply:
Yes. It standartize error payloads of the API and can be use to separate user friendly message from technical message.